src/pal/src/configure.cmake

```diff
@@ -33,7 +33,9 @@ check_include_files(sys/lwp.h HAVE_SYS_LWP_H)
 check_include_files(lwp.h HAVE_LWP_H)
 check_include_files(libunwind.h HAVE_LIBUNWIND_H)
 check_include_files(runetype.h HAVE_RUNETYPE_H)
+set(CMAKE_REQUIRED_FLAGS "-ldl")
```

Comment:
Could you please use CMAKE_REQUIRED_LIBRARIES instead to be in line with other places in this file?
Also, FreeBSD and NetBSD don't have libdl, so this should be conditional (you can use `NOT CMAKE_SYSTEM_NAME STREQUAL FreeBSD AND NOT CMAKE_SYSTEM_NAME STREQUAL NetBSD` condition).